test/e2e/framework/test_context.go

```diff
@@ -283,6 +290,7 @@ func RegisterCommonFlags(flags *flag.FlagSet) {
 	flags.StringVar(&TestContext.ImageServiceEndpoint, "image-service-endpoint", "", "The image service endpoint of cluster VM instances.")
 	flags.StringVar(&TestContext.DockershimCheckpointDir, "dockershim-checkpoint-dir", "/var/lib/dockershim/sandbox", "The directory for dockershim to store sandbox checkpoints.")
 	flags.StringVar(&TestContext.KubernetesAnywherePath, "kubernetes-anywhere-path", "/workspace/k8s.io/kubernetes-anywhere", "Which directory kubernetes-anywhere is installed to.")
+	flags.StringVar(&TestContext.WhitelistedTaint, "whitelist-taint-regexp", `^node-role\.kubernetes\.io/master$`, "Nodes with taints which match this regexp will not block the test framework from starting tests.")
```

Comment:
I think "Whitelist" taints is confusing naming. Instinctively I read it as "Taints that specifically aren't ignored".  A few other names that come to mind:
* non-blocking-taints
* ignored-taints

I'm also wondering why regex? I think most use-cases will be for specific taints like the control plane ones, we should keep it simple and have this a comma delimited list of taint key?

Reply:
Regex was just because in the early incarnation of this issue I created a DD and the only comment I got was that maybe it would be more versatile if we did a regexp in case people have `foo/x` `foo/y` etc. The indication I got was that there may be some people with a large number of taints, though I'm not personally aware of situations with large # of unique taints.

If we had a static list we could simplify some of the logic and add only a single new field to the testContext instead of the string/regexp pair.

I'm fine with a rename on the variable, its not a big issue to me.`NonBlockingTaints` sgtm

Reply:
https://github.com/kubernetes/kubernetes/issues/74282#issuecomment-470217210

@neolit123 Can you elaborate on the need? I don't have a hat in the ring on that point (regexp vs list) but you said it wouldn't be ideal to have to explicitly list them. Do you standby that enough that you'd be against the list implementation? Both @timothysc and @andrewsykim think the list is more intuitive/simple/sufficient.

Reply:
regexp seems more flexible to me.
in terms of naming of the flag, best to exclude it's "type" of the name e.g. `my-flag-int vs my-flag`

Reply:
IMO we should stick to list of taints and **if** the use-case comes up, we can adapt it to handle catch-alls for known prefixes as opposed to regex, for example:
```
--non-blocking-taints="node-role.kubernetes.io/master,node-role.kubernetes.io/special-node,my-cloud.io/*"
```

Reply:
I feel like @andrewsykim has the more compelling approach here:
 - avoids premature generalization
 - simplifies interface presented to the user (which would also simplify things for the implementation)

@neolit123, you're right its more flexible (which is why I originally wrote it that way based on your input) but I feel like in order to get this to move forward this simpler approach may be more appropriate. Do you disagree [enough to want to '/hold' this if we don't use the regexp approach]?

This has now been moved to v1.17 so it will have to stall for a bit, but I'll make the tweak to just be a string and get ready for the un-freeze. Hopefully I can get review/consensus by then and it will go in right away

Reply:
@johnSchnake no strong opinions on my side.
let's defer to the `/test` approvers on the decision here: @timothysc @andrewsykim @oomichi 

> This has now been moved to v1.17 so it will have to stall for a bit

at least, until the 10th of September when code freeze is lifted.

Reply:
👍 Changes made to make this a csv list of strings that is passed around. Consistent with other csv flags, we pass around the whole list as a single string and let the user parse it as needed.